c10/util/Optional.h

```diff
@@ -426,45 +407,44 @@ struct trivially_copyable_optimization_optional_base {
 // section 3.2.3 of the System V ABI document
 // (https://www.uclibc.org/docs/psABI-x86_64.pdf).
 template <class ArrayRefT>
-class arrayref_optional_base {
- public:
+struct arrayref_optional_base {
+  using T = typename ArrayRefT::value_type;
+  using size_type = typename ArrayRefT::size_type;
+
   union storage {
     struct raw {
       // ArrayRef has the invariant that if Data is nullptr then
       // Length must be zero, so this is an unused bit pattern.
-      const void* p = nullptr;
-      size_t sz = 1;
-    } uninitialized_{};
-    ArrayRefT value_;
+      const T* p{nullptr};
+      size_type sz{1};
+    } uninitialized_;
```

Comment:
it's silly, but I think removing the braces here will cause clang-tidy to complain. If it doesn't, great!

Do the braces vs = signs on the preceding two lines make a difference or is it just a style thing?